File: 0209. Minimum Size Subarray Sum.cpp
class Solution
{
public:
    int minSubArrayLen(int target, vector<int>& nums)
    {
        int ans = INT_MAX;
        int sum = 0;
        int r = 0;
        int l = 0;

        while (r < nums.size())
        {
            sum += nums[r];

            while (sum >= target)
            {
                ans = min(ans, r - l + 1);
                sum -= nums[l];
                ++l;
            }

            ++r;
        }

        return ans == INT_MAX ? 0 : ans;
    }
};

File: 0209. Minimum Size Subarray Sum.cs
public class Solution
{
    public int MinSubArrayLen(int target, int[] nums)
    {
        var result = int.MaxValue;
        var sum = 0;
        var r = 0;
        var l = 0;

        while (r < nums.Length)
        {
            sum += nums[r];

            while (sum >= target)
            {
                result = Math.Min(result, r - l + 1);
                sum -= nums[l];
                ++l;
            }

            ++r;
        }

        return result == int.MaxValue ? 0 : result;
    }
}